### PR TITLE
Networking v2: Routers: Parse external_fixed_ips

### PR DESCRIPTION
**Pull request by BenoitKnecht**
For #515

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

https://github.com/openstack/neutron-lib/blob/master/neutron_lib/api/definitions/l3.py#L93-L99

### CODE REVIEW

**Comment by jtopjian** [comment]
@BenoitKnecht Thank you for submitting this. Is there somewhere else in the Neutron code that further defines `external_fixed_ips`? Specifically something that confirms `external_fixed_ips` contains only a `subnet_id` and `ip_address`.

**Comment by BenoitKnecht** [comment]
@jtopjian The only thing I could find is [this description](https://github.com/openstack/neutron-lib/blob/master/api-ref/source/v2/parameters.yaml#L4844-L4848).

**Comment by jtopjian** [comment]
We need to find the piece of code that sets those two attributes. Unfortunately this can sometimes be more difficult than making the PR itself. We're not _trying_ to be difficult, and it's not that I don't believe this PR is correct, but for code review and to have a historical reference to why PRs were done a certain way, we need to find the neutron code.

 From looking at the link you gave, I see this:

```
'type:fixed_ips': None,
```

Is it possible that this is a reference to another schema (maybe in another Neutron repo) for `fixed_ips`?

File: openstack/networking/v2/extensions/layer3/routers/results.go [hunk 1]
```diff
@@ -8,7 +8,15 @@ import (
...
 type GatewayInfo struct {
-	NetworkID string `json:"network_id"`
+	NetworkID        string            `json:"network_id"`
+	ExternalFixedIPs []ExternalFixedIP `json:"external_fixed_ips,omitempty"`
+}
+
+// ExternalFixedIP is the IP address and subnet ID of the external gateway of a
+// router.
+type ExternalFixedIP struct {
+	IP       string `json:"ip_address"`
```
Comment (jtopjian) [inline openstack/networking/v2/extensions/layer3/routers/results.go]: Let's change this to `IPAddress` for [consistency](https://github.com/gophercloud/gophercloud/blob/master/openstack/networking/v2/ports/results.go#L45-L49)

Comment (BenoitKnecht) [reply openstack/networking/v2/extensions/layer3/routers/results.go]: @jtopjian Thanks a lot for digging through Neutron and finding that reference!

I've renamed `IP` to `IPAddress` as requested.

**Comment by jtopjian** [comment]
@BenoitKnecht Looks good! Thanks for submitting this.

**Pull request merged**